pkg/providers/vault/tests/main.go

```diff
@@ -0,0 +1,169 @@
+package main
```

Comment:
I understand this supports testing.  Why does this exist as a separate process?  Is there no way to incorporate this into your tests directly?

Reply:
My understanding is that a standalone Vault dev server is needed to test the token refresh - if you refer it as a separated process. Otherwise the author needs to create a fake Vault server within the process which may be difficult.
However I see the main.go can be converted into test cases as it is referred the same way as other test files in the testing guide as below:
```
"Run the test golang file `go run main.go -token=<token-copied-from-step-2> -window=12 -period=6 -host=localhost`:
```

Reply:
This test is meant to be more like end-2-end functional testing actually.  There are also already several unit tests covered in the code shown in this PR but the Vault server side is mocked not real one in the unit tests.

Reply:
End-to-end (or cross-service acceptance) testing is a TAF concern.  I can't help but feel this shouldn't be here.

Reply:
sure, we can remove this test package and just pass it to TAF related tasks.

Reply:
I like the test as it proves the module gets the work done as expected. And I agree that this is not a unit test, but more like a end-to-end testing. We may need to find a better place for it. I am OK to get the test into the instruction side if not within the project.

Reply:
@tingyuz , maybe just put the contents of that several things on the instruction side of PR? What is the place to put this??

Reply:
Removed from here and let me know if any new place can this put into.